src/common/AriaQueryHandler.ts

```diff
@@ -111,6 +121,29 @@ const queryAllArray = async (
   return jsHandle;
 };
 
+// If multiple waitFor are set up asynchronously, we need to wait for the first
+// one to set up the binding in the page before running the others.
+let settingUpBinding = null;
```

Comment:
Is this a reasonable way to implement a semaphore-like waiting logic between asynchrounous operations?

Reply:
So bindings are set up per world? What would happen if we do `addHandlerToWorld(world1); addHandlerToWorld(world2)`?

Reply:
Ah, good point! In practice, bindings are set up per execution context, but the world is responsible for re-binding when the execution context changes. It's fine to add bindings in two different worlds concurrently (it should happen when doing `waitForSelector` in two different frames), so maybe this code actually synchronizes a bit too much. I'll try pushing it into the worlds instead.

Reply:
Moved to `DOMWorld.ts` to allow asyncronously setting up bindings in multiple worlds.